webapi/CopilotChat/Controllers/ChatController.cs

```diff
@@ -137,27 +138,54 @@ public ChatController(ILogger<ChatController> logger, ITelemetryService telemetr
         return this.Ok(chatSkillAskResult);
     }
 
+    /// <summary>
+    /// Parse plugin auth values from request headers.
+    /// </summary>
+    private Dictionary<string, string> GetPluginAuthHeaders(IHeaderDictionary headers)
+    {
+        // Create a regex to match the headers
+        var regex = new Regex("x-sk-copilot-(.*)-auth", RegexOptions.IgnoreCase);
+
+        // Create a dictionary to store the matched headers and values
+        var openApiSkillsAuthHeaders = new Dictionary<string, string>();
+
+        // Loop through the request headers and add the matched ones to the dictionary
+        foreach (var header in headers)
+        {
+            var match = regex.Match(header.Key);
+            if (match.Success)
+            {
+                // Use the first capture group as the key and the header value as the value
+                openApiSkillsAuthHeaders.Add(match.Groups[1].Value.ToUpperInvariant(), header.Value);
+            }
+        }
+
+        return openApiSkillsAuthHeaders;
+    }
+
     /// <summary>
     /// Register skills with the planner's kernel.
     /// </summary>
-    private async Task RegisterPlannerSkillsAsync(CopilotChatPlanner planner, OpenApiSkillsAuthHeaders openApiSkillsAuthHeaders, ContextVariables variables)
+    private async Task RegisterPlannerSkillsAsync(CopilotChatPlanner planner, Dictionary<string, string> openApiSkillsAuthHeaders, ContextVariables variables)
     {
         // Register authenticated skills with the planner's kernel only if the request includes an auth header for the skill.
 
         // Klarna Shopping
-        if (openApiSkillsAuthHeaders.KlarnaAuthentication != null)
+        if (openApiSkillsAuthHeaders.TryGetValue("KLARNA", out string? KlarnaAuthHeader))
```

Comment:
We no longer has constants for KLARNA, GITHUB, JIRA, GRAPH... ?

Reply:
These were never constants, only their request headers were. `openApiSkillsAuthHeaders` was a class we used to manually deserialize the right headers (see the change in line 74 of ChatController) from the request, but now we parse all request headers dynamically, so `openApiSkillsAuthHeaders` is now a dictionary object, where the key is extracted from the capturing group in this regex: `x-sk-copilot-(.*)-auth`

This is the only place these values are used in the webap. The only other place they would be needed is in the webapp when the Plugins are defined (see PluginState.ts) but we can't share constants across the two servers, so there's no use in defining them as constants here.